v2/index/index.go

```diff
@@ -82,9 +83,8 @@ func Attach(path string, idx Index, offset uint64) error {
 // The written bytes include the index encoding.
 // This can then be read back using index.ReadFrom
 func WriteTo(idx Index, w io.Writer) error {
-	buf := make([]byte, binary.MaxVarintLen64)
-	b := binary.PutUvarint(buf, uint64(idx.Codec()))
-	if _, err := w.Write(buf[:b]); err != nil {
+	b := varint.ToUvarint(uint64(idx.Codec()))
```

Comment:
FYI go-uvarint still has PutUvarint - I think it's slightly better if we're in charge of the []byte allocation, because it's easier for the compiler to put that on our stack and avoid an allocation